File: backend/app/schemas/notification.py
"""
Notification Schemas - 通知Schema定义
"""
from typing import Optional, List
from pydantic import BaseModel, Field


class NotificationBase(BaseModel):
    """通知基础Schema"""
    title: str = Field(..., max_length=255)
    message: str
    type: str = Field(default="info")  # info, warning, error, success
    priority: str = Field(default="medium")  # low, medium, high, urgent
    action_url: Optional[str] = Field(None, max_length=500)


class NotificationCreate(NotificationBase):
    """创建通知"""
    recipient_id: int
    sender_id: Optional[int] = None
    project_id: Optional[int] = None
    issue_id: Optional[int] = None


class NotificationUpdate(BaseModel):
    """更新通知"""
    is_read: Optional[bool] = None
    title: Optional[str] = Field(None, max_length=255)
    message: Optional[str] = None


class NotificationResponse(NotificationBase):
    """通知响应"""
    id: int
    recipient_id: int
    sender_id: Optional[int] = None
    project_id: Optional[int] = None
    issue_id: Optional[int] = None
    is_read: bool
    read_at: Optional[str] = None
    created_at: str
    updated_at: str

    class Config:
        from_attributes = True


class NotificationMarkRead(BaseModel):
    """标记已读"""
    notification_ids: List[int]


class NotificationSummary(BaseModel):
    """通知摘要"""
    total: int
    unread: int
    unread_by_type: dict

File: frontend/src/types/notification.ts
/**
 * Notification Types - 通知类型定义
 */

export type NotificationType = 'info' | 'warning' | 'error' | 'success' | 'assignment' | 'status' | 'comment' | 'reminder'
export type NotificationPriority = 'low' | 'medium' | 'high' | 'urgent'

export interface Notification {
  id: number
  title: string
  message: string
  type: NotificationType
  priority: NotificationPriority
  is_read: boolean
  read_at?: string
  action_url?: string
  recipient_id: number
  sender_id?: number
  project_id?: number
  issue_id?: number
  created_at: string
  updated_at: string
}

export interface NotificationCreate {
  title: string
  message: string
  type?: NotificationType
  priority?: NotificationPriority
  action_url?: string
  recipient_id: number
  sender_id?: number
  project_id?: number
  issue_id?: number
}

export interface NotificationUpdate {
  is_read?: boolean
  title?: string
  message?: string
}

export interface NotificationSummary {
  total: number
  unread: number
  unread_by_type: Record<string, number>
}
